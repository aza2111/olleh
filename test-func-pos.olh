function void switchTurns(Player p1, Player p2):
	if (p1.turn):
		p1.turn = false;
		p2.turn = true;
	else: 
		p2.turn = false;
		p1.turn = true;
	end;
end;
