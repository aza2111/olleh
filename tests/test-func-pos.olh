/*
 * Positive: testing function call
 */

function void printFun():
        print "printing from a function";
end;

printFun();

function int intFun():
        return 2;
end;
int x = intFun();
print x;

function int intArgFun(int input):
        return input / 2;
end;
print intArgFun(x);


function String stringFun(int a1, int a2):
        String ans = "";
        for (a1):
                for (a2):
                        ans = ans + ".";
                end;
        end;
        return ans;
end;
print stringFun(1,3);

print "x before: " + x;
function void manipGlobal():
        x = x - 7;
end;
manipGlobal();
print "x after: " + x;
