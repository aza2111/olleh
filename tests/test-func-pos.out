printing from a function
2
1
...
x before: 2
x during: -5
x after: -5
